docs/block-api.md

```diff
@@ -0,0 +1,99 @@
+# Block API
+
+Blocks are the fundamntal element of the Gutenberg editor. They are the primary way in which plugins and themes can register their own functionality and extend the capabilities of the editor. This document covers the main properties of block registration.
+
+## Register Block Type
+
+Every block starts by registering a new block type definition. The function `registerBlockType` takes two arguments, a block `name` and a block configuration object.
+
+### Block Name
+
+The name for a block is a unique string that identifies a block. Names have to be structured as `namespace/block-name`, where namespace is the name of your plugin or theme.
+
+```js
+// Registering my block with a unique name
+registerBlockType( 'my-plugin/book', {} );
+```
+
+### Block Configuration
+
+A block requires a few properties to be specified before it can be registered successfully. These are defined through a configuration object, which includes the following:
+
+#### Title
+
+This is the display title for your block, which can be translated with our translation functions. The block inserter will show this name.
+
+```js
+// Our data object
+title: 'Book'
+```
+
+#### Category
+
+Blocks are grouped into categories to help users browse and discover them. The core provided categories are `common`, `formatting`, `layout`, `widgets`, and `embeds`.
+
+```js
+// Assigning to the 'layout' category
+category: 'widgets',
+```
+
+#### Icon
```

Comment:
`icon` is optional (though now that I think of it, I'm not sure how well it appears in the inserter)

Reply:
We should offer a generic fallback, probably, and keep it optional.